lib/elixir/lib/calendar/date.ex

```diff
@@ -1159,13 +1159,13 @@ defmodule Date do
 
   defimpl Inspect do
     def inspect(%{calendar: calendar, year: year, month: month, day: day}, _)
-        when year in -9999..9999 do
+        when calendar == Calendar.ISO do
       "~D[" <> calendar.date_to_string(year, month, day) <> suffix(calendar) <> "]"
     end
 
     def inspect(%{calendar: calendar, year: year, month: month, day: day}, _)
-        when calendar == Calendar.ISO do
-      "Date.new!(#{Integer.to_string(year)}, #{Integer.to_string(month)}, #{Integer.to_string(day)})"
+        when year in -9999..9999 do
+      "~D[" <> calendar.date_to_string(year, month, day) <> suffix(calendar) <> "]"
     end
 
     def inspect(%{calendar: calendar, year: year, month: month, day: day}, _) do
```

Comment:
I think we should remove this clause and the guards. Just always invoke `calendar.date_to_string`. :)

Reply:
I was thinking that, but wanted to avoid a situation where a particular calendar implementation is inadvertently relying on year being in `-9999..9999`.